db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -881,10 +931,14 @@ public CostEstimate estimateCost(OptimizablePredicateList predList,
         baseTableRestrictionList.removeAllElements();
         baseTableRestrictionList.countScanFlags();
 
-        currentJoinStrategy.getBasePredicates(predList,baseTableRestrictionList,this);
         /* RESOLVE: Need to figure out how to cache the StoreCostController */
         StoreCostController scc=getStoreCostController(tableDescriptor,cd);
         useRealTableStats=scc.useRealTableStatistics();
+        if (!currentIndexFirstColumnStats.setFrom(cd.getFirstColumnStats())) {
```

Comment:
I would like to guarantee that firstColumnStats is never null in the storeCostController, so better not to check it.  Also, checking it should have no effect because we always do the first pass with disableIndexPrefixIteration == false before setting  disableIndexPrefixIteration to true and re-costing.